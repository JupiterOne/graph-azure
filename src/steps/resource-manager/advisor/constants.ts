import { RelationshipClass } from '@jupiterone/integration-sdk-core';
import { SecurityEntities } from '../security/constants';
import { ANY_SCOPE } from '../constants';

export const AdvisorSteps = {
  RECOMMENDATIONS: 'rm-advisor-recommendations',
  ASSESSMENT_RECOMMENDATION_RELATIONSHIP:
    'rm-advisor-assessment-recommendation-relationships',
  RESOURCE_RECOMMENDATION_RELATIONSHIP:
    'rm-advisor-resource-recommendation-relationships',
};

export const AdvisorEntities = {
  RECOMMENDATION: {
    _type: 'azure_advisor_recommendation',
    _class: ['Finding'],
    resourceName: '[RM] Advisor Recommendation',
  },
};

export const AdvisorRelationships = {
  ASSESSMENT_IDENTIFIED_FINDING: {
    _type: 'azure_assessment_identified_recommendation',
    sourceType: SecurityEntities.ASSESSMENT._type,
    _class: RelationshipClass.IDENTIFIED,
    targetType: AdvisorEntities.RECOMMENDATION._type,
  },
  ANY_RESOURCE_HAS_FINDING: {
    _type: 'azure_resource_has_finding',
    sourceType: ANY_SCOPE,
    _class: RelationshipClass.HAS,
    targetType: AdvisorEntities.RECOMMENDATION._type,
  },
};
