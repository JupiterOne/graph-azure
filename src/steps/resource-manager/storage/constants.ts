import { RelationshipClass } from '@jupiterone/integration-sdk-core';
import { entities as keyvaultEntities } from '../key-vault/constants';

// Step IDs
export const steps = {
  STORAGE_ACCOUNTS: 'rm-storage-resources',
  STORAGE_CONTAINERS: 'rm-storage-containers',
  STORAGE_FILE_SHARES: 'rm-storage-file-shares',
  STORAGE_QUEUES: 'rm-storage-queues',
  STORAGE_TABLES: 'rm-storage-tables',
};

// Graph objects
/**
 * azure_storage_account --HAS--> (azure_storage_container | azure_storage_queue | azure_storage_file_share | azure_storage_table)
 * azure_storage_container --HAS--> azure_storage_blob
 * azure_storage_file_share --HAS--> azure_storage_file_share_directory
 * azure_storage_table --HAS--> azure_storage_table_entity
 */
export const entities = {
  STORAGE_ACCOUNT: {
    _type: 'azure_storage_account',
    _class: ['Service'],
    resourceName: '[RM] Storage Account',
  },
  STORAGE_CONTAINER: {
    _type: 'azure_storage_container',
    _class: ['DataStore'],
    resourceName: '[RM] Storage Container',
  },
  STORAGE_FILE_SHARE: {
    _type: 'azure_storage_file_share',
    _class: ['DataStore'],
    resourceName: '[RM] Storage File Share',
  },
  STORAGE_QUEUE: {
    _type: 'azure_storage_queue',
    _class: ['Queue'],
    resourceName: '[RM] Storage Queue',
  },
  STORAGE_TABLE: {
    _type: 'azure_storage_table',
    _class: ['DataStore', 'Database'],
    resourceName: '[RM] Storage Table',
  },
};

export const relationships = {
  STORAGE_ACCOUNT_HAS_CONTAINER: {
    _type: 'azure_storage_account_has_container',
    sourceType: entities.STORAGE_ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: entities.STORAGE_CONTAINER._type,
  },
  STORAGE_ACCOUNT_HAS_FILE_SHARE: {
    _type: 'azure_storage_account_has_file_share',
    sourceType: entities.STORAGE_ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: entities.STORAGE_FILE_SHARE._type,
  },
  STORAGE_ACCOUNT_HAS_QUEUE: {
    _type: 'azure_storage_account_has_queue',
    sourceType: entities.STORAGE_ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: entities.STORAGE_QUEUE._type,
  },
  STORAGE_ACCOUNT_HAS_TABLE: {
    _type: 'azure_storage_account_has_table',
    sourceType: entities.STORAGE_ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: entities.STORAGE_TABLE._type,
  },
  STORAGE_ACCOUNT_USES_KEY_VAULT: {
    _type: 'azure_storage_account_uses_keyvault_service',
    sourceType: entities.STORAGE_ACCOUNT._type,
    _class: RelationshipClass.USES,
    targetType: keyvaultEntities.KEY_VAULT._type,
  },
};
