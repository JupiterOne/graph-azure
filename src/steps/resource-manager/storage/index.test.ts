import {
  executeStepWithDependencies,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { setupAzureRecording } from '../../../../test/helpers/recording';
import { getStepTestConfigForStep } from '../../../../test/integrationInstanceConfig';
import { steps } from './constants';

let recording: Recording | undefined;

afterEach(async () => {
  await recording?.stop();
});

test('rm-storage-resources', async () => {
  const stepTestConfig = getStepTestConfigForStep(steps.STORAGE_ACCOUNTS);

  recording = setupAzureRecording(
    {
      name: steps.STORAGE_ACCOUNTS,
      directory: __dirname,
    },
    stepTestConfig.instanceConfig,
  );

  const stepResults = await executeStepWithDependencies(stepTestConfig);
  expect(stepResults).toMatchStepMetadata(stepTestConfig);
}, 100_000);

test('rm-storage-containers', async () => {
  const stepTestConfig = getStepTestConfigForStep(steps.STORAGE_CONTAINERS);

  recording = setupAzureRecording(
    {
      name: steps.STORAGE_CONTAINERS,
      directory: __dirname,
    },
    stepTestConfig.instanceConfig,
  );

  const stepResults = await executeStepWithDependencies(stepTestConfig);
  expect(stepResults).toMatchStepMetadata(stepTestConfig);
}, 500_000);

test('rm-storage-file-shares', async () => {
  const stepTestConfig = getStepTestConfigForStep(steps.STORAGE_FILE_SHARES);

  recording = setupAzureRecording(
    {
      name: steps.STORAGE_FILE_SHARES,
      directory: __dirname,
    },
    stepTestConfig.instanceConfig,
  );

  const stepResults = await executeStepWithDependencies(stepTestConfig);
  expect(stepResults).toMatchStepMetadata(stepTestConfig);
}, 200_000);

test('rm-storage-queues', async () => {
  const stepTestConfig = getStepTestConfigForStep(steps.STORAGE_QUEUES);

  recording = setupAzureRecording(
    {
      name: steps.STORAGE_QUEUES,
      directory: __dirname,
    },
    stepTestConfig.instanceConfig,
  );

  const stepResults = await executeStepWithDependencies(stepTestConfig);
  expect(stepResults).toMatchStepMetadata(stepTestConfig);
}, 200_000);

test('rm-storage-tables', async () => {
  const stepTestConfig = getStepTestConfigForStep(steps.STORAGE_TABLES);

  recording = setupAzureRecording(
    {
      name: steps.STORAGE_TABLES,
      directory: __dirname,
    },
    stepTestConfig.instanceConfig,
  );

  const stepResults = await executeStepWithDependencies(stepTestConfig);
  expect(stepResults).toMatchStepMetadata(stepTestConfig);
}, 200_000);
