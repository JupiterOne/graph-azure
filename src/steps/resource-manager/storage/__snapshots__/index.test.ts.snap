// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rm-storage-accounts step 1`] = `
Array [
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "accessTier": "Hot",
          "allowBlobPublicAccess": true,
          "creationTime": 2021-09-20T15:02:41.564Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-20T15:02:41.704Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-20T15:02:41.704Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
          "kind": "StorageV2",
          "location": "eastus",
          "minimumTlsVersion": "TLS1_2",
          "name": "regenerateaccesskeytest",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://regenerateaccesskeytest.blob.core.windows.net/",
            "web": "https://regenerateaccesskeytest.z13.web.core.windows.net/",
          },
          "primaryLocation": "eastus",
          "privateEndpointConnections": Array [
            Object {
              "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest/privateEndpointConnections/regenerateaccesskeytest.a90df093-d49e-4f7b-a354-db550d27e73e",
              "name": "regenerateaccesskeytest.a90df093-d49e-4f7b-a354-db550d27e73e",
              "privateEndpoint": Object {
                "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Network/privateEndpoints/private-endpoint",
              },
              "privateLinkServiceConnectionState": Object {
                "actionRequired": "None",
                "description": "Auto-Approved",
                "status": "Approved",
              },
              "provisioningState": "Succeeded",
              "type": "Microsoft.Storage/storageAccounts/privateEndpointConnections",
            },
          ],
          "provisioningState": "Succeeded",
          "sku": Object {
            "name": "Premium_LRS",
            "tier": "Premium",
          },
          "statusOfPrimary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": true,
    "allowSharedKeyAccess": undefined,
    "blobAnalyticsLoggingDeleteEnabled": undefined,
    "blobAnalyticsLoggingReadEnabled": undefined,
    "blobAnalyticsLoggingWriteEnabled": undefined,
    "blobSoftDeleteEnabled": undefined,
    "blobSoftDeleteRetentionDays": undefined,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1632150161564,
    "displayName": "regenerateaccesskeytest",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://regenerateaccesskeytest.blob.core.windows.net/",
      "https://regenerateaccesskeytest.z13.web.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
    "kind": "StorageV2",
    "lastAccessKeyRegenerationDate": undefined,
    "name": "regenerateaccesskeytest",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": undefined,
    "queueAnalyticsLoggingReadEnabled": undefined,
    "queueAnalyticsLoggingWriteEnabled": undefined,
    "region": "eastus",
    "resourceGroup": "defaultresourcegroup-cus",
    "sku": "Premium_LRS",
    "tableAnalyticsLoggingDeleteEnabled": undefined,
    "tableAnalyticsLoggingReadEnabled": undefined,
    "tableAnalyticsLoggingWriteEnabled": undefined,
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
  },
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_SQL_Resources/providers/Microsoft.Storage/storageAccounts/sqlvab7pt3ww5qhmdw",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "accessTier": "Hot",
          "allowBlobPublicAccess": false,
          "creationTime": 2021-09-23T11:32:50.815Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-23T11:32:50.940Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-23T11:32:50.940Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_SQL_Resources/providers/Microsoft.Storage/storageAccounts/sqlvab7pt3ww5qhmdw",
          "kind": "StorageV2",
          "location": "eastus",
          "minimumTlsVersion": "TLS1_2",
          "name": "sqlvab7pt3ww5qhmdw",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://sqlvab7pt3ww5qhmdw.blob.core.windows.net/",
            "dfs": "https://sqlvab7pt3ww5qhmdw.dfs.core.windows.net/",
            "file": "https://sqlvab7pt3ww5qhmdw.file.core.windows.net/",
            "queue": "https://sqlvab7pt3ww5qhmdw.queue.core.windows.net/",
            "table": "https://sqlvab7pt3ww5qhmdw.table.core.windows.net/",
            "web": "https://sqlvab7pt3ww5qhmdw.z13.web.core.windows.net/",
          },
          "primaryLocation": "eastus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "sku": Object {
            "name": "Standard_LRS",
            "tier": "Standard",
          },
          "statusOfPrimary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
      Object {
        "name": "blobServiceProperties",
        "rawData": Object {
          "blobAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
          "cors": Array [],
          "defaultServiceVersion": undefined,
          "deleteRetentionPolicy": Object {
            "AllowPermanentDelete": "false",
            "days": undefined,
            "enabled": false,
          },
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "staticWebsite": Object {
            "defaultIndexDocumentPath": undefined,
            "enabled": false,
            "errorDocument404Path": undefined,
            "indexDocument": undefined,
          },
        },
      },
      Object {
        "name": "queueServiceProperties",
        "rawData": Object {
          "cors": Array [],
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "queueAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": false,
    "allowSharedKeyAccess": undefined,
    "blobAnalyticsLoggingDeleteEnabled": false,
    "blobAnalyticsLoggingReadEnabled": false,
    "blobAnalyticsLoggingWriteEnabled": false,
    "blobSoftDeleteEnabled": false,
    "blobSoftDeleteRetentionDays": undefined,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1632396770815,
    "displayName": "sqlvab7pt3ww5qhmdw",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://sqlvab7pt3ww5qhmdw.blob.core.windows.net/",
      "https://sqlvab7pt3ww5qhmdw.queue.core.windows.net/",
      "https://sqlvab7pt3ww5qhmdw.table.core.windows.net/",
      "https://sqlvab7pt3ww5qhmdw.file.core.windows.net/",
      "https://sqlvab7pt3ww5qhmdw.z13.web.core.windows.net/",
      "https://sqlvab7pt3ww5qhmdw.dfs.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_SQL_Resources/providers/Microsoft.Storage/storageAccounts/sqlvab7pt3ww5qhmdw",
    "kind": "StorageV2",
    "lastAccessKeyRegenerationDate": undefined,
    "name": "sqlvab7pt3ww5qhmdw",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": false,
    "queueAnalyticsLoggingReadEnabled": false,
    "queueAnalyticsLoggingWriteEnabled": false,
    "region": "eastus",
    "resourceGroup": "stefan_sql_resources",
    "sku": "Standard_LRS",
    "tableAnalyticsLoggingDeleteEnabled": false,
    "tableAnalyticsLoggingReadEnabled": false,
    "tableAnalyticsLoggingWriteEnabled": false,
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_SQL_Resources/providers/Microsoft.Storage/storageAccounts/sqlvab7pt3ww5qhmdw",
  },
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "accessTier": "Hot",
          "allowBlobPublicAccess": true,
          "creationTime": 2021-09-07T13:14:18.446Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-07T13:14:18.555Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-07T13:14:18.555Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
          "kind": "StorageV2",
          "location": "eastus",
          "minimumTlsVersion": "TLS1_2",
          "name": "stefanexamplestorage",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://stefanexamplestorage.blob.core.windows.net/",
            "dfs": "https://stefanexamplestorage.dfs.core.windows.net/",
            "file": "https://stefanexamplestorage.file.core.windows.net/",
            "queue": "https://stefanexamplestorage.queue.core.windows.net/",
            "table": "https://stefanexamplestorage.table.core.windows.net/",
            "web": "https://stefanexamplestorage.z13.web.core.windows.net/",
          },
          "primaryLocation": "eastus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "secondaryEndpoints": Object {
            "blob": "https://stefanexamplestorage-secondary.blob.core.windows.net/",
            "dfs": "https://stefanexamplestorage-secondary.dfs.core.windows.net/",
            "queue": "https://stefanexamplestorage-secondary.queue.core.windows.net/",
            "table": "https://stefanexamplestorage-secondary.table.core.windows.net/",
            "web": "https://stefanexamplestorage-secondary.z13.web.core.windows.net/",
          },
          "secondaryLocation": "westus",
          "sku": Object {
            "name": "Standard_RAGRS",
            "tier": "Standard",
          },
          "statusOfPrimary": "available",
          "statusOfSecondary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
      Object {
        "name": "blobServiceProperties",
        "rawData": Object {
          "blobAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
          "cors": Array [],
          "defaultServiceVersion": undefined,
          "deleteRetentionPolicy": Object {
            "AllowPermanentDelete": "false",
            "days": 7,
            "enabled": true,
          },
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "staticWebsite": Object {
            "defaultIndexDocumentPath": undefined,
            "enabled": false,
            "errorDocument404Path": undefined,
            "indexDocument": undefined,
          },
        },
      },
      Object {
        "name": "queueServiceProperties",
        "rawData": Object {
          "cors": Array [],
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "queueAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": true,
    "allowSharedKeyAccess": undefined,
    "blobAnalyticsLoggingDeleteEnabled": false,
    "blobAnalyticsLoggingReadEnabled": false,
    "blobAnalyticsLoggingWriteEnabled": false,
    "blobSoftDeleteEnabled": true,
    "blobSoftDeleteRetentionDays": 7,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1631020458446,
    "displayName": "stefanexamplestorage",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://stefanexamplestorage.blob.core.windows.net/",
      "https://stefanexamplestorage.queue.core.windows.net/",
      "https://stefanexamplestorage.table.core.windows.net/",
      "https://stefanexamplestorage.file.core.windows.net/",
      "https://stefanexamplestorage.z13.web.core.windows.net/",
      "https://stefanexamplestorage.dfs.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
    "kind": "StorageV2",
    "lastAccessKeyRegenerationDate": undefined,
    "name": "stefanexamplestorage",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": false,
    "queueAnalyticsLoggingReadEnabled": false,
    "queueAnalyticsLoggingWriteEnabled": false,
    "region": "eastus",
    "resourceGroup": "stefan_storage_resources",
    "sku": "Standard_RAGRS",
    "tableAnalyticsLoggingDeleteEnabled": false,
    "tableAnalyticsLoggingReadEnabled": false,
    "tableAnalyticsLoggingWriteEnabled": false,
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
  },
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "creationTime": 2021-09-16T14:50:27.037Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-16T14:50:27.130Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-16T14:50:27.130Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
          "kind": "Storage",
          "location": "centralus",
          "minimumTlsVersion": "TLS1_2",
          "name": "storageaccountstefaafb8",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://storageaccountstefaafb8.blob.core.windows.net/",
            "file": "https://storageaccountstefaafb8.file.core.windows.net/",
            "queue": "https://storageaccountstefaafb8.queue.core.windows.net/",
            "table": "https://storageaccountstefaafb8.table.core.windows.net/",
          },
          "primaryLocation": "centralus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "sku": Object {
            "name": "Standard_LRS",
            "tier": "Standard",
          },
          "statusOfPrimary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
      Object {
        "name": "blobServiceProperties",
        "rawData": Object {
          "blobAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
          "cors": Array [],
          "defaultServiceVersion": undefined,
          "deleteRetentionPolicy": Object {
            "AllowPermanentDelete": "false",
            "days": undefined,
            "enabled": false,
          },
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "staticWebsite": undefined,
        },
      },
      Object {
        "name": "queueServiceProperties",
        "rawData": Object {
          "cors": Array [],
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "queueAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": undefined,
    "allowSharedKeyAccess": undefined,
    "blobAnalyticsLoggingDeleteEnabled": false,
    "blobAnalyticsLoggingReadEnabled": false,
    "blobAnalyticsLoggingWriteEnabled": false,
    "blobSoftDeleteEnabled": false,
    "blobSoftDeleteRetentionDays": undefined,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1631803827037,
    "displayName": "storageaccountstefaafb8",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://storageaccountstefaafb8.blob.core.windows.net/",
      "https://storageaccountstefaafb8.queue.core.windows.net/",
      "https://storageaccountstefaafb8.table.core.windows.net/",
      "https://storageaccountstefaafb8.file.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
    "kind": "Storage",
    "lastAccessKeyRegenerationDate": undefined,
    "name": "storageaccountstefaafb8",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": false,
    "queueAnalyticsLoggingReadEnabled": false,
    "queueAnalyticsLoggingWriteEnabled": false,
    "region": "centralus",
    "resourceGroup": "stefan_appservices_group",
    "sku": "Standard_LRS",
    "tableAnalyticsLoggingDeleteEnabled": false,
    "tableAnalyticsLoggingReadEnabled": false,
    "tableAnalyticsLoggingWriteEnabled": false,
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
  },
]
`;
